junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/LifecycleMethodExecutionExceptionHandlerTests.java

```diff
@@ -66,39 +66,26 @@ void resetStatics() {
 		throwExceptionAfterAll = true;
 		handlerCalls.clear();
 
-		SwallowExceptionHandler.beforeAllCalls = 0;
-		SwallowExceptionHandler.beforeEachCalls = 0;
-		SwallowExceptionHandler.afterEachCalls = 0;
-		SwallowExceptionHandler.afterAllCalls = 0;
-
-		RethrowExceptionHandler.beforeAllCalls = 0;
-		RethrowExceptionHandler.beforeEachCalls = 0;
-		RethrowExceptionHandler.afterEachCalls = 0;
-		RethrowExceptionHandler.afterAllCalls = 0;
-
-		ConvertExceptionHandler.beforeAllCalls = 0;
-		ConvertExceptionHandler.beforeEachCalls = 0;
-		ConvertExceptionHandler.afterEachCalls = 0;
-		ConvertExceptionHandler.afterAllCalls = 0;
-
-		UnrecoverableExceptionHandler.beforeAllCalls = 0;
-		UnrecoverableExceptionHandler.beforeEachCalls = 0;
-		UnrecoverableExceptionHandler.afterEachCalls = 0;
-		UnrecoverableExceptionHandler.afterAllCalls = 0;
-
-		ShouldNotBeCalledHandler.beforeAllCalls = 0;
-		ShouldNotBeCalledHandler.beforeEachCalls = 0;
-		ShouldNotBeCalledHandler.afterEachCalls = 0;
-		ShouldNotBeCalledHandler.afterAllCalls = 0;
+		SwallowExceptionHandler.callCounter.reset();
+
+		RethrowExceptionHandler.callCounter.reset();
+
+		ConvertExceptionHandler.callCounter.reset();
+
+		UnrecoverableExceptionHandler.callCounter.reset();
+
+		ShouldNotBeCalledHandler.callCounter.reset();
 	}
 
 	@Test
 	void classLevelExceptionHandlersRethrowException() {
 		LauncherDiscoveryRequest request = request().selectors(selectClass(RethrowingTestCase.class)).build();
 		EngineExecutionResults executionResults = executeTests(request);
 
-		assertEquals(1, RethrowExceptionHandler.beforeAllCalls, "Exception should handled in @BeforeAll");
-		assertEquals(1, RethrowExceptionHandler.afterAllCalls, "Exception should handled in @AfterAll");
+		assertEquals(1, RethrowExceptionHandler.callCounter.getBeforeAllCalls(),
```

Comment:
With the introduction of the `HandlerCallCounter` there is no need to store the counts as a field in the handler. You could now declare a `HandlerCallCounter rethrowCounter` as a static field on `LifecycleMethodExecutionExceptionHandlerTests` and reference that everywhere. I.e.

```
assertEquals(1, rethrowCounter.getBeforeAllCalls(), "Exception should handled in @BeforeAll");
```

With some thoughtfully chosen method and field names this would reduce the verbosity of the code a bit and keep the whole assertion on a single line.